src/Helpers/OAuth2/OAuthConfig.php

```diff
@@ -224,6 +232,38 @@ public function setDefaultScopes(array $defaultScopes): static
         return $this;
     }
 
+    /**
+     * Set the request modifier callable which can be used to modify the request being sent
+     *
+     * @param callable(\Saloon\Contracts\Request): (void) $requestModifier
+     * @return $this
+     */
+    public function setRequestModifier(callable $requestModifier): static
+    {
+        $this->requestModifier = $requestModifier;
+
+        return $this;
+    }
+
+    /**
+     * Invoke the OAuth2 config request modifier
+     *
+     * @param \Saloon\Contracts\Request $request
+     * @return \Saloon\Contracts\Request
+     */
```

Comment:
In this case, we even return the very same request, just possibly mutated.

```suggestion
    /**
     * Invoke the OAuth2 config request modifier
     *
     * @template TRequest of \Saloon\Contracts\Request
     *
     * @param TRequest $request
     * @return TRequest
     */
```